test/steps/kogitoapp.go

```diff
@@ -20,23 +20,21 @@ import (
 	"github.com/cucumber/godog"
 	"github.com/cucumber/godog/gherkin"
 	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
-	"github.com/kiegroup/kogito-cloud-operator/pkg/util"
 	"github.com/kiegroup/kogito-cloud-operator/test/framework"
 	"github.com/rdumont/assistdog"
 )
 
 var assist = assistdog.NewDefault()
 
-// RegisterCliSteps register all CLI steps existing
 func registerKogitoAppSteps(s *godog.Suite, data *Data) {
 	// Deploy steps
-	s.Step(`^"([^"]*)" deploy quarkus example service "([^"]*)" with native "([^"]*)"$`, data.deployQuarkusExampleServiceWithNative)
-	s.Step(`^"([^"]*)" deploy quarkus example service "([^"]*)" with native "([^"]*)" and labels$`, data.deployQuarkusExampleServiceWithNativeAndLabels)
-	s.Step(`^"([^"]*)" deploy quarkus example service "([^"]*)" with native "([^"]*)" and persistence$`, data.deployQuarkusExampleServiceWithNativeAndPersistence)
-	s.Step(`^"([^"]*)" deploy quarkus example service "([^"]*)" with native "([^"]*)" and persistence and events$`, data.deployQuarkusExampleServiceWithNativeAndPersistenceAndEvents)
-	s.Step(`^"([^"]*)" deploy spring boot example service "([^"]*)"$`, data.deploySpringBootExampleService)
-	s.Step(`^"([^"]*)" deploy spring boot example service "([^"]*)" with persistence$`, data.deploySpringBootExampleServiceWithPersistence)
-	s.Step(`^"CR" deploy service from example file "([^"]*)"$`, data.deployServiceFromExampleFile)
+	s.Step(`^Deploy quarkus example service "([^"]*)" with native ((?:enabled|disabled))$`, data.deployQuarkusExampleServiceWithNative)
+	s.Step(`^Deploy quarkus example service "([^"]*)" with native ((?:enabled|disabled)) and labels$`, data.deployQuarkusExampleServiceWithNativeAndLabels)
+	s.Step(`^Deploy quarkus example service "([^"]*)" with native ((?:enabled|disabled)) and persistence$`, data.deployQuarkusExampleServiceWithNativeAndPersistence)
+	s.Step(`^Deploy quarkus example service "([^"]*)" with native ((?:enabled|disabled)) and persistence and events$`, data.deployQuarkusExampleServiceWithNativeAndPersistenceAndEvents)
```

Comment:
Do we need a non-capturing group inside the capturing one? What if it was just a one capturing group, i.e. `(enabled|disabled)`?

Reply:
yes we do in fact. just tested before without the double `(` and it does not catch the value.
seems that `(?:enabled|disabled)` is just the choice but does not catch the value to be used as method parameter
`((?:enabled|disabled))` does ;)

Reply:
And why not to use `(enabled|disabled)` ?

Reply:
That's why I suggested `(enabled|disabled)` because `(?:enabled|disabled)` is not capturing so no wonder it doesn't capture anything :)

Reply:
yes better like that... will update

Reply:
updated